File: frontend/src/types/character.ts
export type CharacterData = {
  id: string;
  ownerEmail: string;

  name: string;
  nickname?: string;
  title?: string;
  age: number;

  primaryRace?: string;
  race?: string;

  characterLevel?: number;
  level?: number;

  strength: number;
  dexterity: number;
  constitution: number;
  mind: number;
  attunement: number;

  maxHp: number;
  currentHp: number;
  actionPoints: number;
  initiative: number;
};

File: backend/TabletopRPG.API/Models/Character.cs
namespace TabletopRPG.API.Models
{
    public class Character
    {
        // Profile Info
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public string Name { get; set; } = string.Empty;
        public string Nickname { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public int Age { get; set; }
        public string PrimaryRace { get; set; } = string.Empty;
        public int CharacterLevel { get; set; } = 1;

        // Core Health Stats
        public int CurrentHp { get; set; }
        public int MaxHp { get; set; }

        // Primary Eteria Stats
        public int Strength { get; set; }
        public int Dexterity { get; set; }
        public int Constitution { get; set; }

        // Mental Stats
        public int Mind { get; set; }
        public int Attunement { get; set; }

        // Maneuver & Derived Stats
        public int ActionPoints { get; set; }
        public int Initiative { get; set; }

        // Helper Method: Automated Stat Calculation
        public void CalculateDerivedStats()
        {
            // Automated formula based on primary attributes
            this.MaxHp = (this.Constitution * 10) + (this.CharacterLevel * 5);
            this.CurrentHp = this.MaxHp;
            this.ActionPoints = 2 + (this.Dexterity / 2);
            this.Initiative = this.Dexterity + this.Mind;
        }
    }
}